src/utils/isSameDay.js

```diff
@@ -2,5 +2,9 @@ import moment from 'moment';
 
 export default function isSameDay(a, b) {
   if (!moment.isMoment(a) || !moment.isMoment(b)) return false;
-  return a.isSame(b, 'day');
+  // Compare least significant, most likely to change units first
+  // Moment's isSame clones moment inputs and is a tad slow
```

Comment:
By passing "day" we're telling moment to throw away time, so for the "day" code path it seems like it would be an improvement for them as well.